### PR TITLE
Add temperature shutdown

### PR DESCRIPTION
**Pull request by VanDavv**
This PR adds a temperature shutdown mechanism that raises Overheat error if the device gets over 105* celsius

### CODE REVIEW

File: depthai_demo.py [hunk 1]
```diff
@@ -374,7 +374,11 @@ def _logMonitorCallback(self, msg: dai.LogMessage):
         if msg.level == dai.LogLevel.CRITICAL:
             print(f"[CRITICAL] [{msg.time.get()}] {msg.payload}", file=sys.stderr)
             sys.stderr.flush()
-            self.error = OverheatError(msg.payload)
+            temperature = self._device.getChipTemperature()
+            if temperature.average > 100:
```
Comment (themarpe) [inline depthai_demo.py:378]: Modify to any temp reading > 105.0C

**Comment by themarpe** [comment]
I assume you've raised the OverheatError specifically so that Sentry will pick it up as is, correct?

**Comment by VanDavv** [comment]
Yes, also custom exception can be caught by users who extend the demo further (not to raise a regular RuntimeError when we can distinguish this error)

File: depthai_demo.py [hunk 2]
```diff
@@ -360,6 +367,19 @@ def stop(self, *args, **kwargs):
...
 
+    def canRun(self):
+        return hasattr(self, "_device") and not self._device.isClosed()
+
+    def _logMonitorCallback(self, msg: dai.LogMessage):
+        if msg.level == dai.LogLevel.CRITICAL:
+            print(f"[CRITICAL] [{msg.time.get()}] {msg.payload}", file=sys.stderr)
+            sys.stderr.flush()
+            temperature = self._device.getChipTemperature()
+            if temperature.average > 105:
```
Comment (themarpe) [inline depthai_demo.py:378]: Check if any temperatures is higher than 105C, otherwise it might trigger too late

**Review by themarpe — CHANGES_REQUESTED** [review]

**Review by themarpe — APPROVED** [review]

**Pull request merged**